includes/sanitizers/class-amp-blacklist-sanitizer.php

```diff
@@ -93,24 +102,88 @@ private function sanitize_a_attribute( $node, $attribute ) {
 			// rev removed from HTML5 spec, which was used by Jetpack Markdown.
 			$node->removeAttribute( $attribute_name );
 		} elseif ( 'target' === $attribute_name ) {
-		       if ( '_blank' === $attribute->value || '_new' === $attribute->value ) {
-			       // _new is not allowed; swap with _blank
-			       $node->setAttribute( $attribute_name, '_blank' );
-		       } else {
-			       // only _blank is allowed
-			       $node->removeAttribute( $attribute_name );
-		       }
+			// _blank is the only allowed value and it must be lowercase.
+			// replace _new with _blank and others should simply be removed.
+			$old_value = strtolower( $attribute->value );
+			if ( '_blank' === $old_value || '_new' === $old_value ) {
+				// _new is not allowed; swap with _blank
+				$node->setAttribute( $attribute_name, '_blank' );
+			} else {
+				// only _blank is allowed
+				$node->removeAttribute( $attribute_name );
+			}
+		}
+	}
+
+	private function validate_a_node( $node ) {
+		// Get the href attribute
+		$href = $node->getAttribute( 'href' );
+
+		// If no href is set and this isn't an anchor, it's invalid
+		if ( empty( $href ) ) {
+			$name_attr = $node->getAttribute( 'name' );
+			if ( ! empty( $name_attr ) ) {
+				// No further validation is required
+				return true;
+			} else {
+				return false;
+			}
+		}
+
+		// If this is an anchor link, just return true
+		if ( 0 === strpos( $href, '#' ) ) {
+			return true;
+		}
+
+		// If the href starts with a '/', append the home_url to it for validation purposes.
+		if ( 0 === stripos( $href, '/' ) ) {
+			$href = untrailingslashit( get_home_url() ) . $href;
```

Comment:
Yes, in theory but I still want to check that the relative URL is in a valid format. I think this bit of extra validation is fine.